packages/mobile/src/web3/saga.ts

```diff
@@ -343,6 +343,7 @@ export function* unlockAccount(account: string) {
     }
 
     const pincode = yield call(getPincode)
+    navigateBack()
```

Comment:
I dont't think this belongs here. We use `unlockAccount` in `getConnectedUnlockedAccount` which is used in many places. I don't think navigating back is always the right behavior.

Reply:
Well, previously it was navigation back all the time it was asking for a PIN(PTAL here https://github.com/celo-org/celo-monorepo/pull/2648/files#diff-fdbb05a987fa942d9270541910265e9c in methods `onPressConfirm` and `onPressCancel`). Then when we introduced "Lock wallet with PIN" and I changed it, and forget to update it here. As far as I got when `getConnectedUnlockedAccount` is called and PIN is not cached - the PIN window appears from nowhere and then it goes to nowhere, when user has entered correct PIN or pressed back.

Reply:
Correct, had addressed it.